src/gda/topology.cpp

```
@@ -386,8 +386,9 @@ namespace rocshmem
       // Query the number of IBV devices
       int numIbvDevices = 0;
       ibv_device** deviceList = ibv_get_device_list(&numIbvDevices);
+      CHECK_NNULL(deviceList, "ibv_get_device_list");
```

Is it not going to abort here instead of printing the error mesage line 458?

---

Not nessiarly, from the man page:

> ibv_get_device_list() returns the array of available RDMA devices, or sets errno and returns NULL if the request fails. If no devices are found then num_devices is set to 0, and non-NULL is returned.

Checking null and the num_devices is checking two different situations